### ISSUE TITLE
Add counterpart to bash's `ls -h` to Get-ChildItem

### ISSUE THREAD

**Issue by WillPittenger** [PowerShell/PowerShell#16724]
### Summary of the new feature / enhancement

Currently, when Get-ChildItem returns a file size, it's in bytes—even if the size is several terabytes.  bash lets users specify the `-h` switch to ls.  In Powershell, it's more complex as Get-ChildItem doesn't know anything about the data.  Adding the switch would cause the method to prefer human readable sizes.  The switch would look for a property named `Length` and replace that in the select with `LengthAsStr` (if that exists in the class) which would be assumed to be human readable.

The second step would be to arrange with the authors of `System.Io.FileInfo` and `System.Io.DirectoryInfo` to add `LengthAsStr`.

I considered saying that Get-ChildItem itself could attempt to convert the field, but `Length` might be in something other than bytes, say characters.

**Comment by iSazonov** [comment]
You can create custom view for Format-* cmdlets.

**Comment by jhoneill** [comment]
See https://github.com/PowerShell/PowerShell/issues/16712#issuecomment-1007955825 
That has a link to a format file which does it. 

Having said that putting thousand separators into the numbers even without formatting as KB / MB /GB /TB would make things much easier to read.

**Comment by itsamemarkus** [comment]
I'd really like to see that too. I don't know when it became acceptable to have 20+ line scripts or cmdlets for basic, newbie friendly outputs.

**Comment by sharky98** [comment]
> I'd really like to see that too. I don't know when it became acceptable to have 20+ line scripts or cmdlets for basic, newbie friendly outputs.

I would rephrase that part more like: I don't know when it became acceptable to have 20+ line scripts or cmdlets for basic ~~, newbie~~ friendly outputs ***that needs to be copied over and over on all machines and session the user is using***.

For reference, `ls -h` seems to have been added in 1997 (https://github.com/coreutils/coreutils/commit/9c59f1dcc4da71a09a781d975918d17d1bd78d88) by using `human.c` from gnulib (https://github.com/coreutils/gnulib/commit/1bbd61b95a3b89e7ca993880d5c44ca391d54f0a). Oh, and reading the commit, this was inspired from the `du` commands, which got its `-h` flag in 1996 (https://github.com/coreutils/coreutils/commit/744d31d57a943db9c7e5074540d7121d84b1a50f). So, this human-readable feature was available 6 years before the first thought on PowerShell were made (Monad, 2002), 10 years before the PowerShell V1.0 release (in 2006) and 19 years before the first "Initial Commit" on this repo (2015, https://github.com/PowerShell/PowerShell/commit/81eae75cc9dd14765389c239f6b708c0ad780ade). This is a bit of rant, I admit, but it does shows that a very old tool build in the day when Linux was usable only by geek and command line already had user-friendly features, whereas a very modern tool build to be user-friendly doesn't even provide such basic user-friendly feature.
